# Hack allowing to bundle whole unpacked extensions without their manifests
# breaking the packaging process.

--- a/python/mozbuild/mozpack/packager/__init__.py	2014-10-04 19:03:00.000000000 +0000
+++ b/python/mozbuild/mozpack/packager/__init__.py	2014-10-06 12:07:03.000000000 +0000
@@ -246,7 +246,10 @@
         '''
         assert not self._closed
         if is_manifest(path):
-            self._add_manifest_file(path, file)
+            if "chrome.manifest" in path and "extensions" in path:
+                self._file_queue.append(self.formatter.add, path, file)
+            else:
+                self._add_manifest_file(path, file)
         elif path.endswith('.xpt'):
             self._queue.append(self.formatter.add_interfaces, path, file)
         else:
